ersilia/core/tracking.py

```diff
@@ -536,7 +537,27 @@ def track(self, input, result, meta):
         self.docker_client = SimpleDocker()
         json_dict = {}
         input_data = read_csv(input)
-        result_data = read_csv(result)
+        # Create a temporary file to store the result if it is a generator
+        if isinstance(result, types.GeneratorType):
+
+            # Ensure EOS/tmp directory exists
+            tmp_dir = os.path.join(EOS, "tmp")
+            if not os.path.exists(tmp_dir):
+                os.makedirs(tmp_dir, exist_ok=True)
+
+            # Create a temporary file to store the generator output
```

Comment:
Reword similarly as above.

Reply:
Do you mean I should reword to "create a temporary file to store the result if it is a generator" same as above?